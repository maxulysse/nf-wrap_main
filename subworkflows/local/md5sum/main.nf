// MD5SUM

include { MD5SUM } from '../../../modules/nf-core/md5sum/main.nf'

workflow RUN_MD5SUM {
    take:
    input // channel: [mandatory] [ meta, file ] ]

    main:
    MD5SUM(input)

    emit:
    checksum = MD5SUM.out.checksum
    versions = MD5SUM.out.versions
}
