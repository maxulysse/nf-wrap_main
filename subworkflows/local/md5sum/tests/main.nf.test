nextflow_workflow {

    name "Test Workflow RUN_MD5SUM"
    script "../main.nf"
    workflow "RUN_MD5SUM"
    tag "nf-wrap/md5sum"

    test("MD5SUM") {

        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.checksum).match("checksum")},
                { assert snapshot(workflow.out.versions).match("versions")}
            )
        }
    }
}
