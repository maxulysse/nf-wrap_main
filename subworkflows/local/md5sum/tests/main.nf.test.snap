{
    "versions": {
        "content": [
            "h"
        ],
        "timestamp": "2023-11-27T12:19:09.277731"
    },
    "checksum": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1.fastq.gz.md5:md5,63a3f187d06ae5cffe77fcd39bbdd63a"
                ]
            ]
        ],
        "timestamp": "2023-11-27T12:08:52.133951"
    }
}